src/monster.cpp

```diff
@@ -931,16 +931,15 @@ void Monster::onThinkDefense(uint32_t interval)
 			}
 
 			Monster* summon = Monster::createMonster(summonBlock.name);
-			if (summon) {
-				const Position& summonPos = getPosition();
-
-				addSummon(summon);
-
-				if (!g_game.placeCreature(summon, summonPos, false, summonBlock.force)) {
-					removeSummon(summon);
-				} else {
+			if (summon){
+				if (g_game.placeCreature(summon, getPosition(), false, summonBlock.force)) {
+					summon->setDropLoot(false);
+					summon->setLossSkill(false);
+					summon->setMaster(this);
 					g_game.addMagicEffect(getPosition(), CONST_ME_MAGIC_BLUE);
 					g_game.addMagicEffect(summon->getPosition(), CONST_ME_TELEPORT);
+				} else {
+					delete summon;
```

Comment:
placeCreature returns false if is not placed, so there is no reference, and as it was created just before, I see no problem on deleting it